File: Problems/SimplifyPath.java
class Solution {
    public String simplifyPath(String path) {
        Stack<String> aux = new Stack<>();
        for(String i : path.split("/")){
            if(i.equals("..")){
                if(!aux.isEmpty())
                    aux.pop();
            }else if(i.length() > 0 && !i.equals(".")){
                aux.add(i);                              
            }
        }
        return "/" + String.join("/",aux);
    }
}

File: Problems/SimplifyPath.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        result = []
        path = path.split('/')
        for i in path:
            if i:
                if i==".":
                    continue
                elif i=="..":
                    if result:
                        result.pop()
                else:
                    result.append(i)
        return "/"+"/".join(result)
